app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -610,6 +611,21 @@ public void onLauncherButtonClicked(View launcherButton) {
 
     @Override
     public boolean dispatchTouchEvent(MotionEvent ev) {
+        if (ev.getAction() == MotionEvent.ACTION_DOWN) {
+            View v = getCurrentFocus();
+            if (v instanceof SearchEditText) {
+                SearchEditText edit = ((SearchEditText) v);
+                Rect outR = new Rect();
+                edit.getGlobalVisibleRect(outR);
+                Boolean isKeyboardOpen = !outR.contains((int)ev.getRawX(), (int)ev.getRawY());
+                if (isKeyboardOpen) {
+                    edit.clearFocus();
+                    InputMethodManager imm = (InputMethodManager) this.getSystemService(Context.INPUT_METHOD_SERVICE);
+                    imm.hideSoftInputFromWindow(edit.getWindowToken(), 0);
+                }
```

Comment:
Maybe it's sufficient to remove only this part that is hiding keyboard?

Reply:
Yes, I just removed the part that hides the keyboard.  I left the part that changes focus. I tried changing the conditions of hiding the keyboard with `HIDE_NOT_ALWAYS` and `HIDE_IMPLICIT_ONLY`, but the keyboard is opened by both `SHOW_IMPLICIT` and `SHOW_FORCED`, so that does not work.

Reply:
https://github.com/Neamar/KISS/pull/2282

Reply:
#2282 now checks if the correct view is active before trying to hide the keyboard. This seems to be the best solution because the user can still select outside of the Search Bar on the desktop to hide the keyboard without interfering with the menu displayed by long pressing on search results.

Reply:
> > #2282 now checks if the correct view is active before trying to hide the keyboard. This seems to be the best solution because the user can still select outside of the Search Bar on the desktop to hide the keyboard without interfering with the menu displayed by long pressing on search results.
> 
> can you please check the build? I made a build but it isn't fixed. still happens on your cursor fix branch.

I am unable to reproduce the issue with #2282 applied.

Reply:
I am unsure of this is an issue with Miui 14, or perhaps just newer versions of Android. You could try building the debug version of KISS, but I'm not very sure how much help that would be. It would help to know of anyone with other versions of Android are having the same issue, but without being able to reproduce the issue anymore myself, it might be tough for me to fix this properly.

Reply:
@t-e-s-tweb The objective is to hide the keyboard when user clicks outside the searchedit view.  When user clicks outside the searchedit view,  then `isKeyboarOpen` should be true. The current code is doing what is intended.

Can you please try this solution:
https://github.com/syphyr/KISS/commit/af4f9b2b44dbe2f9d7a60590e88a649d266cc608

Reply:
@t-e-s-tweb Please look at https://github.com/Neamar/KISS/pull/2285 and leave a comment there.

Reply:
> this introduces a new bug. without this commit if keyboard is open and you long click on the list then the app menu appears, you can click and launch the app or do anything from menu.
> 
> however after this, the moment you touch the app, the keyboard closes and app list moves to bottom which interferes with opening app menus and even launching. please look into it. thanks.

Same here: it happens on an old android 10 (lineage os) and also on android 14 (avd)
As soon as keyboard is hidden because search is not focused any more, everything moves and menu is cut off.

Reply:
> > this introduces a new bug. without this commit if keyboard is open and you long click on the list then the app menu appears, you can click and launch the app or do anything from menu.
> > however after this, the moment you touch the app, the keyboard closes and app list moves to bottom which interferes with opening app menus and even launching. please look into it. thanks.
> 
> Same here: it happens on an old android 10 (lineage os) and also on android 14 (avd) As soon as keyboard is hidden because search is not focused any more, everything moves and menu is cut off.

Does just removing `edit.clearFocus();` fix it?

Reply:
I noticed that removing `edit.clearFocus();` on my devices causes the bug to show up.  I'm starting to think that the problem is that the parent view is not focus-able on newer versions of android.  Perhaps adding `"android:focusable="true"` to  app/src/main/res/layout/main.xml would fix it for newer versions of android.